src/lib/utils/meta_tables/meta_indexes_table.cpp

```diff
@@ -0,0 +1,38 @@
+#include "meta_indexes_table.hpp"
+#include "storage/index/segment_index_type.hpp"
+
+#include "hyrise.hpp"
+
+namespace opossum {
+
+MetaIndexesTable::MetaIndexesTable()
+    : AbstractMetaTable(TableColumnDefinitions{{"index_name", DataType::String, false},
+                                               {"index_type", DataType::String, false},
+                                               {"table_name", DataType::String, false},
+                                               {"column_names", DataType::String, false},
+                                               }) {}
+
+const std::string& MetaIndexesTable::name() const {
+  static const auto name = std::string{"indexes"};
+  return name;
+}
+
+std::shared_ptr<Table> MetaIndexesTable::_on_generate() const {
+  auto output_table = std::make_shared<Table>(_column_definitions, TableType::Data, std::nullopt, UseMvcc::Yes);
+
+  for (const auto& [table_name, table] : Hyrise::get().storage_manager.tables()) {
+    for (auto index : table->indexes_statistics()) {
+      std::string column_names ("");
+      for (auto col_id : index.column_ids) {
+        column_names += " " + table->column_name(col_id);
+      }
+      output_table->append({pmr_string{index.name}, static_cast<pmr_string>(segment_index_type_to_string(index.type)),
```

Comment:
segment_index_type_to_string is not necessary here. magic_enum::enum_name can be used instead